pandas/tests/frame/test_indexing.py

```diff
@@ -1548,6 +1548,25 @@ def test_setitem_single_column_mixed_datetime(self):
         # pytest.raises(
         #    Exception, df.loc.__setitem__, ('d', 'timestamp'), [nan])
 
+    def test_setitem_mixed_datetime(self):
+        # GH 9336
+        expected = DataFrame({'a': [0, 0, 0, 0, 13, 14],
+                              'b': [pd.datetime(2012, 1, 1),
+                                    1,
+                                    'x',
+                                    'y',
+                                    pd.datetime(2013, 1, 1),
+                                    pd.datetime(2014, 1, 1)]})
+        df = pd.DataFrame(np.zeros((6, 2), dtype=int), columns=['a', 'b'])
```

Comment:
this is failing because you are specify the dtype directly I think, just use 
``pd.DataFrame(0, columns = list('ab'), index=range(6))``